test/benchmarks/datadog/metric_handler.py

```diff
@@ -0,0 +1,167 @@
+        valid_tags = []
+        if tags is not None:
+            for tag in tags:
+                is_dataset_size_tag = tag.value in DatasetSizeTags.values()
+                is_reader_model_tag = tag.value in ReaderModelTags.values()
+                is_retriever_model_tag = tag.value in RetrieverModelTags.values()
+                is_document_store_tag = tag.value in DocumentStoreModelTags.values()
+                is_benchmark_type_tag = tag.value in BenchmarkType.values()
+
+                if (
+                    is_dataset_size_tag
+                    or is_reader_model_tag
+                    or is_retriever_model_tag
+                    or is_document_store_tag
+                    or is_benchmark_type_tag
+                ):
+                    valid_tags.append(tag)
+                elif tag == NoneTag.none:
+                    pass
+                else:
+                    # Log invalid tags as errors
+                    LOGGER.error(f"Tag is not a valid dataset or environment tag: tag={tag}")
+
+        return valid_tags
+
+
+class IndexingDocsPerSecond(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.indexing.docs_per_second"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingExactMatchMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.exact_match"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingF1Metric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.f1_score"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingRecallMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.recall"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingMapMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.map"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingSecondsPerQueryMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.seconds_per_query"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class MetricsAPI:
+    @retry(
+        retry=retry_if_exception(AssertionError),  # type: ignore
+        wait=wait_fixed(5),
+        stop=stop_after_attempt(3),
+        reraise=True,
+    )
+    def send_custom_dd_metric(self, metric: CustomDatadogMetric) -> dict:
+        options = {
+            "api_key": os.getenv("DD_API_KEY"),
+            "app_key": os.getenv("DD_APP_KEY"),
```

Comment:
This is not required for submitting metrics, see https://docs.datadoghq.com/api/latest/metrics/#submit-metrics